libs/wallet/src/connect-dialog/connect-dialog.tsx

```diff
@@ -28,15 +29,37 @@ type WalletType = 'gui' | 'cli' | 'hosted';
 
 export interface VegaConnectDialogProps {
   connectors: Connectors;
+  onChangeOpen?: (open: boolean) => void;
+}
+
+export const useVegaWalletDialogStore = create<VegaWalletDialogStore>(
+  (set) => ({
+    dialogOpen: false,
+    updateDialogOpen: (open: boolean) => set({ dialogOpen: open }),
+  })
+);
+
+interface VegaWalletDialogStore {
   dialogOpen: boolean;
-  setDialogOpen: (isOpen: boolean) => void;
+  updateDialogOpen: (open: boolean) => void;
 }
 
 export const VegaConnectDialog = ({
   connectors,
-  dialogOpen,
-  setDialogOpen,
+  onChangeOpen,
 }: VegaConnectDialogProps) => {
+  const { dialogOpen, updateDialogOpen } = useVegaWalletDialogStore(
```

Comment:
As mentioned above. Move useVegaWalletDialogStore to trading app.
We can have component in trading app which will wrap VegaConnectDialog and will use useVegaWalletDialogStore to pass `open` and `onOpenChange` props

Reply:
So do you really want me to remove my change of vega wallet dialog state from here?